h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/hadoop/h2odriver.java

```diff
@@ -1496,7 +1542,48 @@ else if (rv != 0) {
       return 0;
     }
 
-    System.out.println("(Note: Use the -disown option to exit the driver after cluster formation)");
+    if (client) {
+      if (flatfileContent == null)
+        throw new IllegalStateException("ERROR: flatfile should have been created by now.");
+
+      final File flatfile = File.createTempFile("h2o", "txt");
+      flatfile.deleteOnExit();
+
+      Writer w = new BufferedWriter(new FileWriter(flatfile));
+      boolean flatfileCreated = false;
+      try {
+        w.write(flatfileContent);
+        w.close();
+        flatfileCreated = true;
+      } catch (IOException e) {
+        e.printStackTrace();
+      } finally {
+        try {
+          w.close();
+        } catch (IOException suppressed) { /* ignore */ }
+      }
+
+      if (!flatfileCreated) {
+        System.out.println("ERROR: Failed to write flatfile.");
+        System.exit(1);
+      }
+
+      String[] generatedClientArgs = new String[]{
+              "-client",
+              "-flatfile", flatfile.getAbsolutePath(),
+              "-md5skip",
+              "-user_name", userName
```

Comment:
In theory, you should setup`-name` which is consistent with name setup for backend machines (in `    addMapperArg(conf, "-name", jobtrackerName);`)